src/python/pants/backend/google_cloud_function/python/target_types.py

```diff
@@ -244,6 +250,17 @@ class PythonGoogleCloudFunction(Target):
         f"See {doc_url('python-google-cloud-function')}."
     )
 
+    def validate(self) -> None:
+        if (
+            self[PythonGoogleCloudFunctionRuntime].value is None
```

Comment:
N.B.: I started off making this a mutex, but realized we have no business saying the lambdex has to be single-platform. Multi is fine if that's what you want. The approach then is to ensure at least 1, the rest being left up to the user.